### PR TITLE
Make telopmode mode not block the robot

### PR DESCRIPTION
**Pull request by MaxiMaerz**
Tested and it works.
Question is do we want this or do we want a parameter

### CODE REVIEW

**Review by Eman7C7 — COMMENTED** [review]
Make it a parameter and make it spam zero by default anyway. I think it's safer this way

**Review by Eman7C7 — CHANGES_REQUESTED** [review]
forgot to "request changes" 😛

**Comment by MaxiMaerz** [comment]
The default will stay the same, however if the 'force_stop_on_idle' is set:
* we will not send 0 unless we have sent some other velocity before
* also we will only publish once

File: README.md [hunk 1]
```diff
@@ -142,10 +145,11 @@ teleop:                        # Parameters for the Teleoperate mode.
   default_increment: 0.1       # Default velocity increment when pressing a key.
   increment_step: 0.1          # Step for increasing the velocity increment.
   cmd_vel_topic: cmd_vel       # Topic on which to publish the velocity commands.
+  force_publish_on_idel: false # If false keep publishing 0 velocities, only publish once otherwise
```
Comment (Eman7C7) [inline README.md:148]: idle typo

File: src/app_modes/teleoperate.rs [hunk 2]
```diff
@@ -36,12 +38,24 @@ impl Teleoperate {
             current_velocities: initial_velocities,
             increment: config.default_increment,
             increment_step: config.increment_step,
+            force_stop_when_idle: config.force_stop_when_idle,
+            has_published_zero_once: true, // Initialize to 0 so the robot is not stopped when entering the mode
```
Comment (Eman7C7) [inline src/app_modes/teleoperate.rs:42]: initialize to true (in the comment)

**Review by Eman7C7 — CHANGES_REQUESTED** [review]

**Comment by MaxiMaerz** [comment]
Should we make the boolean has_published_zero_once a counter?

File: README.md (hunk 1 above)
Comment (mgrrx) [reply README.md:148]: the name of the parameter is cryptic.
I suggest keep_publishing_cmd_vel_when_idle

Comment (carzum) [reply README.md:148]: publish_cmd_vel_when_idle, or publish_when_idle as its already in the cmd_vel namespace?

File: src/app_modes/teleoperate.rs [hunk 3]
```diff
@@ -63,11 +77,24 @@ impl AppMode for Teleoperate {
...
     fn run(&mut self) {
-        let mut vel_cmd = rosrust_msg::geometry_msgs::Twist::default();
-        vel_cmd.linear.x = self.current_velocities.x;
-        vel_cmd.linear.y = self.current_velocities.y;
-        vel_cmd.angular.z = self.current_velocities.theta;
-        self.cmd_vel_pub.send(vel_cmd).unwrap();
+        // If the velocity is reset to 0 only publish it once
+        // this prevents the robot from being blocked if the
+        // app mode is not closed
+        if  self.force_stop_when_idle
```
Comment (carzum) [inline src/app_modes/teleoperate.rs:83]: i would prefer to not only do it for 0 velocities but for all velocities, ie: send a zero command if no button was pressed for x seconds (5?) and stop publishing

**Review by carzum — CHANGES_REQUESTED** [review]

File: src/app_modes/teleoperate.rs (hunk 3 above)
Comment (MaxiMaerz) [reply src/app_modes/teleoperate.rs:83]: Not so sure here, I think most implementations would assume that the machine should stop.

**Review by carzum — APPROVED** [review]

**Review by Eman7C7 — APPROVED** [review]
Nice! Works perfectly for me

**Pull request merged**